completed assignment 2

rmart300 opened the pull request.

natasha-aleksandrova commented on students/rmart300/lesson04/dict_lab.py:
```diff
@@ -0,0 +1,67 @@
+    for v in person_dict.values():
+        print(v)
+
+    print('Is cake a key in the dict? {}'.format(str('cake' in person_dict)))
```
note that it is not necessary to cast to `str`, this will work just fine:
`print('Is cake a key in the dict? {}'.format('cake' in person_dict))`

natasha-aleksandrova commented on students/rmart300/lesson04/dict_lab.py:
```diff
@@ -0,0 +1,67 @@
+    print('python set :' + str(py_set))
+    print('marathon_set :' + str(mar_set))
+    print('union of the two sets: {}'.format(py_set.union(mar_set)))
+    print('intersection of the two sets: {}'.format(py_set.intersection(mar_set)))
```
nice work on dicts and sets!

natasha-aleksandrova commented on students/rmart300/mailroom/mailroom.py:
```diff
@@ -25,9 +27,10 @@ def send_thank_you():
+    amount_list = donation_dict[name] if name in donation_dict else [] 
+    amount_list.append(amount)
+    donation_dict[name] = amount_list    
+    donor_dict[name] = { 'firstname': name.split(' ')[0], 'lastname': name.split(' ')[1] }
```
no need for spaces around `{}`

natasha-aleksandrova commented on students/rmart300/mailroom/mailroom.py:
```diff
@@ -38,31 +41,47 @@ def create_report():
+       donation_num = 1
+       for donation_amt in amount_list:      
+           message = f"Dear {first_name} {last_name},\n\n    Thank you for your very kind donation of ${donation_amt}.\n\n    It will be put to very good use.\n\n    Sincerely,\n        -The Team"
+           f = open(f"{letter_directory}{first_name}_{last_name}_{donation_num}.txt",'w')
```
recommended to use `with` statement when working with files, example:
```
with open(filename) as fh:
    fh.write(message)
```

natasha-aleksandrova commented on students/rmart300/mailroom/mailroom.py:
```diff
@@ -38,31 +41,47 @@ def create_report():
-    
+
+def write_letters():
+   for donor in donor_dict:
```
instead of accessing by key later it is more pythonic to iterator over keys and values:
```
for donor, donor_data in donor_dict.items():
```

natasha-aleksandrova commented on students/rmart300/mailroom/mailroom.py:
```diff
@@ -38,31 +41,47 @@ def create_report():
+       amount_list = donation_dict[donor]
+       donation_num = 1
+       for donation_amt in amount_list:      
+           message = f"Dear {first_name} {last_name},\n\n    Thank you for your very kind donation of ${donation_amt}.\n\n    It will be put to very good use.\n\n    Sincerely,\n        -The Team"
```
this line is a bit too long... there are couple options to clean this up, my personal preference is using this style:
```
    message = (
       f "Dear {first_name}... \n"
        "Thank you...'"
    )
```

natasha-aleksandrova commented on students/rmart300/mailroom/mailroom.py:
```diff
@@ -38,31 +41,47 @@ def create_report():
+           f = open(f"{letter_directory}{first_name}_{last_name}_{donation_num}.txt",'w')
+           f.write(message)
+           f.close()
+           donation_num += 1
```
I don't think this is used?

natasha-aleksandrova commented on students/rmart300/trigram/trigram.py:
```diff
@@ -0,0 +1,45 @@
+
+if __name__ == '__main__':
+   
+    book_file = sys.argv[1]
```
nice one 🙂

natasha-aleksandrova commented on students/rmart300/trigram/trigram.py:
```diff
@@ -0,0 +1,45 @@
+                else:
+                    trigram_dict[digram] = []
+
+def write_book():
```
this function isn't being used?

natasha-aleksandrova commented on students/rmart300/trigram/trigram.py:
```diff
@@ -0,0 +1,45 @@
+    book = open(book_file, 'r'):
+    digram = ''
+    for line in book:
+        line.strip()
```
this line will have no effect since strings are immutable, if you do want this to take effect you can do:
`words = line.strip().split()`

natasha-aleksandrova commented on students/rmart300/trigram/trigram.py:
```diff
@@ -0,0 +1,45 @@
+
+def read_book(book_file):
+
+    book = open(book_file, 'r'):
```
always make sure to close the file.. and instead of having to worry about that use `with` 🙂

natasha-aleksandrova commented on students/rmart300/trigram/trigram.py:
```diff
@@ -0,0 +1,45 @@
+                if digram in triagram_dict:
+                    word_list = trigram_dict[digram]
+                    word_list.append(words[i+1])
+                    trigram_dict[digram] = wordlist
```
note that you can simplify lines 16-18 to: `trigram_dict[digram].append(words[i + 1])`

natasha-aleksandrova commented on students/rmart300/trigram/trigram.py:
```diff
@@ -0,0 +1,45 @@
+    
+    out = open('my_book.txt','w')
+    max_length = 200
+    while cur_length < max_length:
```
`cur_length` is not defined

rmart300 commented:
Yes,trigram assignment is not finished. Just checked in what i had

On Sat, Feb 17, 2018 at 8:42 PM Natasha <notifications@github.com> wrote:

> *@natasha-aleksandrova* approved this pull request.
>
> Nice work on assignments!
>
> It seems that trigram assignment is not fully finished?
> ------------------------------
>
> In students/rmart300/lesson04/dict_lab.py
> <https://github.com/UWPCE-PythonCert-ClassRepos/Self_Paced-Online/pull/149#discussion_r168939416>
> :
>
> > +def dictionary1(person_dict):
> +    """function to test basic functionality of dictionaries"""
> +    print(person_dict)
> +    person_dict.pop('cake')
> +    print(person_dict)
> +    person_dict['fruit']='Mango'
> +    print(person_dict)
> +    print('keys in dict')
> +    for k in person_dict:
> +        print(k)
> +
> +    print('values in dict')
> +    for v in person_dict.values():
> +        print(v)
> +
> +    print('Is cake a key in the dict? {}'.format(str('cake' in person_dict)))
>
> note that it is not necessary to cast to str, this will work just fine:
> print('Is cake a key in the dict? {}'.format('cake' in person_dict))
> ------------------------------
>
> In students/rmart300/lesson04/dict_lab.py
> <https://github.com/UWPCE-PythonCert-ClassRepos/Self_Paced-Online/pull/149#discussion_r168939441>
> :
>
> > +    print(s3)
> +    print("s4")
> +    print(s4)
> +
> +    print("Is s3 a subset of s2? {}".format(s3.issubset(s2)))
> +    print("Is s4 a subset of s2? {}".format(s4.issubset(s2)))
> +
> +def set2():
> +    """function to test sets and frozensets"""
> +    py_set = set(list('python'))
> +    py_set.add('i')
> +    mar_set = frozenset('marathon')
> +    print('python set :' + str(py_set))
> +    print('marathon_set :' + str(mar_set))
> +    print('union of the two sets: {}'.format(py_set.union(mar_set)))
> +    print('intersection of the two sets: {}'.format(py_set.intersection(mar_set)))
>
> nice work on dicts and sets!
> ------------------------------
>
> In students/rmart300/mailroom/mailroom.py
> <https://github.com/UWPCE-PythonCert-ClassRepos/Self_Paced-Online/pull/149#discussion_r168939465>
> :
>
> > @@ -25,9 +27,10 @@ def send_thank_you():
>                  print(donor)
>
>      amount = float(input("Provide a donation amount:"))
> -    donation_list = donation_dict[name] if name in donation_dict else []
> -    donation_list.append(amount)
> -    donation_dict[name] = donation_list
> +    amount_list = donation_dict[name] if name in donation_dict else []
> +    amount_list.append(amount)
> +    donation_dict[name] = amount_list
> +    donor_dict[name] = { 'firstname': name.split(' ')[0], 'lastname': name.split(' ')[1] }
>
> no need for spaces around {}
> ------------------------------
>
> In students/rmart300/mailroom/mailroom.py
> <https://github.com/UWPCE-PythonCert-ClassRepos/Self_Paced-Online/pull/149#discussion_r168939486>
> :
>
> >          donation_average = round(donation_total/donation_count,2)
>          data_row = "{0:20}  ${1:>15}   {2:>10}   ${3:>15}".format(donor, str(donation_total), str(donation_count), str(donation_average))
>          print(data_row)
> -
> -
> +
> +def write_letters():
> +   for donor in donor_dict:
> +       temp_dict = donor_dict[donor]
> +       first_name = temp_dict['firstname']
> +       last_name = temp_dict['lastname']
> +       amount_list = donation_dict[donor]
> +       donation_num = 1
> +       for donation_amt in amount_list:
> +           message = f"Dear {first_name} {last_name},\n\n    Thank you for your very kind donation of ${donation_amt}.\n\n    It will be put to very good use.\n\n    Sincerely,\n        -The Team"
> +           f = open(f"{letter_directory}{first_name}_{last_name}_{donation_num}.txt",'w')
>
> recommended to use with statement when working with files, example:
>
> with open(filename) as fh:
>     fh.write(message)
>
> ------------------------------
>
> In students/rmart300/mailroom/mailroom.py
> <https://github.com/UWPCE-PythonCert-ClassRepos/Self_Paced-Online/pull/149#discussion_r168939500>
> :
>
> >          donation_average = round(donation_total/donation_count,2)
>          data_row = "{0:20}  ${1:>15}   {2:>10}   ${3:>15}".format(donor, str(donation_total), str(donation_count), str(donation_average))
>          print(data_row)
> -
> -
> +
> +def write_letters():
> +   for donor in donor_dict:
>
> instead of accessing by key later it is more pythonic to iterator over
> keys and values:
>
> for donor, donor_data in donor_dict.items():
>
> ------------------------------
>
> In students/rmart300/mailroom/mailroom.py
> <https://github.com/UWPCE-PythonCert-ClassRepos/Self_Paced-Online/pull/149#discussion_r168939523>
> :
>
> >          donation_average = round(donation_total/donation_count,2)
>          data_row = "{0:20}  ${1:>15}   {2:>10}   ${3:>15}".format(donor, str(donation_total), str(donation_count), str(donation_average))
>          print(data_row)
> -
> -
> +
> +def write_letters():
> +   for donor in donor_dict:
> +       temp_dict = donor_dict[donor]
> +       first_name = temp_dict['firstname']
> +       last_name = temp_dict['lastname']
> +       amount_list = donation_dict[donor]
> +       donation_num = 1
> +       for donation_amt in amount_list:
> +           message = f"Dear {first_name} {last_name},\n\n    Thank you for your very kind donation of ${donation_amt}.\n\n    It will be put to very good use.\n\n    Sincerely,\n        -The Team"
>
> this line is a bit too long... there are couple options to clean this up,
> my personal preference is using this style:
>
>     message = (
>        f "Dear {first_name}... \n"
>         "Thank you...'"
>     )
>
> ------------------------------
>
> In students/rmart300/mailroom/mailroom.py
> <https://github.com/UWPCE-PythonCert-ClassRepos/Self_Paced-Online/pull/149#discussion_r168939554>
> :
>
> > -
> -
> +
> +def write_letters():
> +   for donor in donor_dict:
> +       temp_dict = donor_dict[donor]
> +       first_name = temp_dict['firstname']
> +       last_name = temp_dict['lastname']
> +       amount_list = donation_dict[donor]
> +       donation_num = 1
> +       for donation_amt in amount_list:
> +           message = f"Dear {first_name} {last_name},\n\n    Thank you for your very kind donation of ${donation_amt}.\n\n    It will be put to very good use.\n\n    Sincerely,\n        -The Team"
> +           f = open(f"{letter_directory}{first_name}_{last_name}_{donation_num}.txt",'w')
> +           f.write(message)
> +           f.close()
> +           donation_num += 1
>
> I don't think this is used?
> ------------------------------
>
> In students/rmart300/trigram/trigram.py
> <https://github.com/UWPCE-PythonCert-ClassRepos/Self_Paced-Online/pull/149#discussion_r168939576>
> :
>
> > +    while cur_length < max_length:
> +        digram = random.choice(trigram_dict.keys())
> +
> +        while digram in trigram_dict:
> +            num_words_used = trigram_usage_dict[digram]
> +            word_list = trigram_dict[digram]
> +            third_word = word_list[num_words_used]
> +            trigram_usage_dict[diagram] = num_words_used + 1
> +            out.write(digram + ' ' + trigram)
> +
> +        out.write('.')
> +
> +
> +if __name__ == '__main__':
> +
> +    book_file = sys.argv[1]
>
> nice one 🙂
> ------------------------------
>
> In students/rmart300/trigram/trigram.py
> <https://github.com/UWPCE-PythonCert-ClassRepos/Self_Paced-Online/pull/149#discussion_r168939587>
> :
>
> > +    digram = ''
> +    for line in book:
> +        line.strip()
> +        words = line.split()
> +        for i in range(len(words)):
> +            if i > 1:
> +                digram = f"{words[i-1]} {words[i]}"
> +            if i < len(words)-1:
> +                if digram in triagram_dict:
> +                    word_list = trigram_dict[digram]
> +                    word_list.append(words[i+1])
> +                    trigram_dict[digram] = wordlist
> +                else:
> +                    trigram_dict[digram] = []
> +
> +def write_book():
>
> this function isn't being used?
> ------------------------------
>
> In students/rmart300/trigram/trigram.py
> <https://github.com/UWPCE-PythonCert-ClassRepos/Self_Paced-Online/pull/149#discussion_r168939615>
> :
>
> > @@ -0,0 +1,45 @@
> +
> +trigram_dict={}
> +
> +def read_book(book_file):
> +
> +    book = open(book_file, 'r'):
> +    digram = ''
> +    for line in book:
> +        line.strip()
>
> this line will have no effect since strings are immutable, if you do want
> this to take effect you can do:
> words = line.strip().split()
> ------------------------------
>
> In students/rmart300/trigram/trigram.py
> <https://github.com/UWPCE-PythonCert-ClassRepos/Self_Paced-Online/pull/149#discussion_r168939629>
> :
>
> > @@ -0,0 +1,45 @@
> +
> +trigram_dict={}
> +
> +def read_book(book_file):
> +
> +    book = open(book_file, 'r'):
>
> always make sure to close the file.. and instead of having to worry about
> that use with 🙂
> ------------------------------
>
> In students/rmart300/trigram/trigram.py
> <https://github.com/UWPCE-PythonCert-ClassRepos/Self_Paced-Online/pull/149#discussion_r168939655>
> :
>
> > +
> +def read_book(book_file):
> +
> +    book = open(book_file, 'r'):
> +    digram = ''
> +    for line in book:
> +        line.strip()
> +        words = line.split()
> +        for i in range(len(words)):
> +            if i > 1:
> +                digram = f"{words[i-1]} {words[i]}"
> +            if i < len(words)-1:
> +                if digram in triagram_dict:
> +                    word_list = trigram_dict[digram]
> +                    word_list.append(words[i+1])
> +                    trigram_dict[digram] = wordlist
>
> note that you can simplify lines 16-18 to: trigram_dict[digram].append(words[i
> + 1])
> ------------------------------
>
> In students/rmart300/trigram/trigram.py
> <https://github.com/UWPCE-PythonCert-ClassRepos/Self_Paced-Online/pull/149#discussion_r168939730>
> :
>
> > +            if i < len(words)-1:
> +                if digram in triagram_dict:
> +                    word_list = trigram_dict[digram]
> +                    word_list.append(words[i+1])
> +                    trigram_dict[digram] = wordlist
> +                else:
> +                    trigram_dict[digram] = []
> +
> +def write_book():
> +    trigram_usage_dict = {}
> +    for digram in trigram_dict:
> +        trigram_usage_dict[digram] = 0
> +
> +    out = open('my_book.txt','w')
> +    max_length = 200
> +    while cur_length < max_length:
>
> cur_length is not defined
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/UWPCE-PythonCert-ClassRepos/Self_Paced-Online/pull/149#pullrequestreview-97391605>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ALoVw_DIEwdyzQOarJOqR7DG9QbxzZmHks5tV6oigaJpZM4SJhY->
> .
>
-- 
Ross Martin
704.840.5077
